common/components/GasSlider/GasSlider.tsx

```diff
@@ -79,6 +89,15 @@ class GasSlider extends React.Component<Props, State> {
   private toggleAdvanced = () => {
     this.setState({ showAdvanced: !this.state.showAdvanced });
   };
+
+  private handleGasPriceInput = (raw: string) => {
+    const gasBn = new BN(raw);
+    const value = gasBn.mul(new BN(Units.gwei));
+    this.setState({
+      gasPrice: { raw, value }
+    });
```

Comment:
@HenryNguyen5 is this OK for fudging the component for 200ms? Or did I miss any other factors that go into gasPrice throughout the sagas?

Reply:
The only race condition i can think of is if the user changes gas price and then generates the transaction within 200ms. This would cause the transaction to be generated with the old gas price. If you want to prevent this situation from ever happening I recommend debouncing it within a saga with the `throttle` effect. This would also require you to make an action creator / action type for a "pending" gas price input, which would disable the `GenerateTransaction` component, similar to how the `isNetworkRequestPending` selector works.

Reply:
A practical solution could be bringing down the throttle to 25/50ms, as in practice I don't see users being able to generate a TX quicker than that.

The solution @HenryNguyen5 mentioned does seem to be robust, but with the tradeoff of additional engineering time/complexity.

Reply:
I like Henry's idea, I'll give that a shake. You can hold off on merging until I implement that.

Reply:
@HenryNguyen5 this is done, though I'll admit it ended up being a lot more code than I wanted. I ended up housing the redux state under `network` because 1. adding a field to `state.transaction.fields` that wasn't an object broke some stuff, and 2. it's more inline with tracking network requests than it is tracking field values. If you have suggestions, please let me know.